### PR TITLE
[golang] fixes to allow for better env templating

### PR DESCRIPTION
**Pull request by GrantSheehan**
Updates helm chart to allow multiline templates and updates the example
to point to a more likely example secret path, and a template that can
take any number of env vars contained in the `env` secret and render
them appropriately.

### CODE REVIEW

**Comment by GrantSheehan** [comment]
Proof it works as intended:

```
gsheehan@ubuntu ~/g/h/c/golang ❯❯❯ kubectl exec -it gsheehan-golang-57765f6855-lbbkz ash                                                   fix/golang/fix-env-templating ◼
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
Defaulted container "app" out of: app, vault-agent, vault-agent-init (init)
/ $ cat /vault/secrets/env
NEWRELIC_LICENSE=[redacted]/ $
```

File: charts/golang/values.yaml [hunk 1]
```diff
@@ -165,9 +165,11 @@ vault:
   enabled: false
   secrets: []
     # - name: env
-    #   path: "secret/data/env"
+    #   path: "devops/go-demo/prod/secret/data/env"
     #   value: |-
-    #     export NEWRELIC_LICENSE="{{ .Data.data.NEWRELIC_LICENSE }}"
+    #     {{- range $k, $v := .Data.data -}}
+    #        {{ $k }}={{ $v }}
+    #     {{ end -}}
```
Comment (ellisio) [inline charts/golang/values.yaml:172]: Should we just default to this if `value` is not supplied? That way we use this as a default, but allow them to override their template contents if they desire?

Comment (GrantSheehan) [reply charts/golang/values.yaml:172]: I like the idea including the template in the values.yaml file template and (hopefully) give someone new or unfamiliar with it the clue that you can template the content of a secret however they'd like, but I'm fine with either approach.

Comment (ellisio) [reply charts/golang/values.yaml:172]: @GrantSheehan You can set a default to "-", which docs to explain what is going on. For example:

```
    # The below example will use a standardized template that injects a file formatted in a way that
    # allows libraries like Viper, or dotenv, to load the file at runtime.
    # - name: env
    #   path: "devops/go-demo/prod/secret/data/env"

    # The below example shows you how to build your own template value for the injected Vault file.
    #
    # - name: env
    #   path: "devops/go-demo/prod/secret/data/env"
    #   value: |-
    #     {{- range $k, $v := .Data.data -}}
    #        {{ $k }}={{ $v }}
    #     {{ end -}}
```

Comment (GrantSheehan) [reply charts/golang/values.yaml:172]: I added a default for the template if it's omitted, what do you think?

**Review by ellisio — APPROVED** [review]

**Pull request merged**